src/libai/nn/NeuralNetwork.java

```diff
@@ -188,41 +234,41 @@ public static double euclideanDistance2(double[] a, double[] b) {
 
 	/**
 	 * Calculates the square Euclidean distance between two column matrix.
+	 * <p><i>NOTE:</i> Assertions of the dimensions are made with {@code assert}
+	 * statement. You must enable this on runtime to be effective.</p>
 	 *
 	 * @param a Matrix a.
 	 * @param b Matrix b.
 	 * @return The square Euclidean distance.
 	 */
 	public static double euclideanDistance2(Matrix a, Matrix b) {
-		try {
-			double sum = 0;
-			for (int i = 0; i < a.getRows(); i++) {
-				double diff = (a.position(i, 0) - b.position(i, 0));
-				sum += diff * diff;
-			}
-			return sum;
-		} catch (RuntimeException e) {
-			System.out.println("a: " + a);
-			System.out.println("\nb: " + b);
-			throw e;
+		double sum = 0;
```

Comment:
I was fairly certain I had written them... apparently I didn't (or at least didn't commited it), I'll add it.